tonic/src/transport/server/mod.rs

```diff
@@ -236,6 +236,23 @@ impl Server {
         Router::new(self.clone(), svc)
     }
 
+    /// Create a router with the `S` typed service as the first service.
+    ///
+    /// This will clone the `Server` builder and create a catch-all router that will
+    /// route around different services.
+    ///
+    /// Catchall services are expected to handle all requests they receive (as opposed to a
+    /// NamedService). As they should be stacked ahead of any services specified with `add_service`
+    /// or `add_optional_service`.
+    pub fn add_catchall_service<S>(&mut self, svc: S) -> Router<S, Unimplemented>
```

Comment:
I was hoping we could just change `Router<S, Unimplemented>` to be something like `Router<S, CatchAll>`, what do you think?